python_modules/libraries/dagster-aws/dagster_aws/s3/sensor.py

```diff
@@ -1,43 +1,100 @@
-from typing import Optional
+from datetime import datetime
+from typing import Any, List, Optional
 
 import boto3
 import dagster._check as check
+from dagster._annotations import deprecated
+
+from dagster_aws._stubs import ObjectTypeDef
 
 
 class ClientException(Exception):
     pass
 
 
-def get_s3_keys(
+def get_objects(
     bucket: str,
     prefix: str = "",
     since_key: Optional[str] = None,
-    s3_session: Optional[boto3.Session] = None,
-):
+    since_last_modified: Optional[str] = None,
+    client=None,
+) -> List[ObjectTypeDef]:
+    """Retrieves a list of object keys in S3 for a given `bucket`, `prefix`, and filter option.
+
+    Args:
+        bucket (str): s3 bucket
+        prefix (str): s3 object prefix
+        since_key (Optional[str]): retrieve objects after the modified date of this key
+        since_last_modified (Optional[str]): retrieve objects after this timestamp
+        client (Optional[boto3.Client]): s3 client
+
+    Returns:
+        List of object keys in S3.
+
+    """
     check.str_param(bucket, "bucket")
     check.str_param(prefix, "prefix")
     check.opt_str_param(since_key, "since_key")
+    check.opt_str_param(since_last_modified, "since_last_modified")
 
-    if not s3_session:
-        s3_session = boto3.client("s3", use_ssl=True, verify=True)
+    if not client:
+        client = boto3.client("s3")
 
-    if not s3_session:
+    if not client:
         raise ClientException("Failed to initialize s3 client")
 
-    paginator = s3_session.get_paginator("list_objects_v2")  # type: ignore
-    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix)
+    paginator = client.get_paginator("list_objects_v2")
+    page_iterator = paginator.paginate(Bucket=bucket, Prefix=prefix, Delimiter="")
 
-    contents = []
+    objects: List[ObjectTypeDef] = []
     for page in page_iterator:
-        contents.extend(page.get("Contents", []))
+        contents = page.get("Contents", [])
+        objects.extend([ObjectTypeDef(obj) for obj in contents])
```

Comment:
Generally `TypedDict` are intended to be used as type annotations for dictionaries. 
They aren't supposed to be instantiated. 
I suggest this to be changed to `cast(ObjectTypeDef, obj)`.

Reply:
Fixed, thanks for informing me of that. 👍